pkg/controller/cronjob/utils.go

```diff
@@ -292,6 +292,16 @@ func IsJobFinished(j *batchv1.Job) bool {
 	return isFinished
 }
 
+// IsJobSucceeded returns whether a job has completed successfully.
+func IsJobSucceeded(j *batchv1.Job) bool {
```

Comment:
Could we use https://github.com/kubernetes/kubernetes/blob/1b7e819fbd227118c00d5d273b45ce04d2f336db/staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go#L102?

Reply:
I'm fine with the function `IsJobSucceeded` but maybe we can reuse this function?

Reply:
> Could we use `IsStatusConditionTrue`?

not really, because job uses its dedicated JobCondition type rather than metav1.Condition. I have no idea why, but this is where we are, maybe as a follow up we could have a designated package for utility functions for JobConditions

Reply:
> Sure, maybe as a follow up we could have a designated package for utility functions for JobConditions. We need to find a good place where to put the functions. cc @deads2k

The pkg/controller/ package used here looks like a very practical place to put this sort of method.

Reply:
Ok, but then should we have a dependency from ` pkg/controller/cronjob` to `pkg/controller/job`, or maybe have a dedicated package, like ` pkg/controller/job_utils`?

Reply:
We already have some identical functions [Job](https://github.com/kubernetes/kubernetes/blob/47737eca1e5daf486d6a2ef4c642655531e776d9/pkg/controller/job/utils.go#L26) and [CronJob](https://github.com/kubernetes/kubernetes/blob/47737eca1e5daf486d6a2ef4c642655531e776d9/pkg/controller/cronjob/utils.go#L290), so it feels like a separate "good-first-time" issue I could open, but it would be nice to have some suggestion for folks to take. cc @deads2k

Reply:
Having `cronjob` depend on `job` sounds reasonable at first blush (that's where I would look).  We do also have `pkg/controller/util/<foo>` with a few foos already.

Reply:
Any preference @alculquicondor?

Reply:
I like `pkg/controller/job/util`

Reply:
Ok, I opened https://github.com/kubernetes/kubernetes/issues/123445. We may then put then `IsJobSucceeded` there. I think it is better not to mix this PR with the cleanup.